proto/tendermint/abci/types.proto

```diff
@@ -295,6 +374,21 @@ message TxResult {
   ResponseDeliverTx result = 4 [(gogoproto.nullable) = false];
 }
 
+// TxAction contains App-provided information on what to do with a transaction that is part of a raw proposal
+enum TxAction {
```

Comment:
Another option to consider would be to nest this enum inside `TransactionRecord`, and then the `ACTION_` prefix wouldn't be necessary. YMMV.

Reply:
I had this question when writing this, but I read [here](https://earthly.dev/blog/backward-and-forward-compatibility/) that "grouping messages" is not good for forward compatibility.

After reading your comment, I went back to that blog and I realize I had misunderstood, confusing "nesting" and "grouping".

So, I guess that, if nesting is safe w.r.t. compatibility, it's probably cleaner to do as you suggest and move the enum inside the message, and remove the "ACTION_" prefixes.

Do you agree with this?

Reply:
Yeah, nesting is fine. It's not crucial, the main thing is that it avoids stutter in the generated code. The `ACTION_UNKNOWN` enumerator at top-level would look like `abcipb.TxAction_ACTION_UNKNOWN` in Go or `abci::TxAction::ACTION_UNKNOWN` in C++.

Whether or not we nest it, it would be nice to drop the `ACTION_` prefixes. And if the enum is _only_ used within `TransactionRecord`, that would also be a helpful hint to the reader. YMMV.